diff --git a/pkgs/by-name/ni/nix-index-unwrapped/package.nix b/pkgs/by-name/ni/nix-index-unwrapped/package.nix
index 97ce3cea6..467cb1f40 100644
--- a/pkgs/by-name/ni/nix-index-unwrapped/package.nix
+++ b/pkgs/by-name/ni/nix-index-unwrapped/package.nix
@@ -10,13 +10,13 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "nix-index";
-  version = "0.1.9";
+  version = "0.1.9-0fc3804";
 
   src = fetchFromGitHub {
     owner = "nix-community";
     repo = "nix-index";
-    rev = "v${version}";
-    hash = "sha256-kOVmgST/D3zNOcGVu1ReuPuVrUx41iRK4rs59lqYX74=";
+    rev = "0fc38040a22a08052103d0fbbafd67ac54165f2b";
+    hash = "";
   };
 
   cargoHash = "sha256-0yrTPrxN/4TOALqpQ5GW7LXKisc8msx3DvEpg8uO+IQ=";
