diff --git a/nixos/modules/system/boot/loader/grub/install-grub.pl b/nixos/modules/system/boot/loader/grub/install-grub.pl
index 6f0f62546..b5448dad1 100644
--- a/nixos/modules/system/boot/loader/grub/install-grub.pl
+++ b/nixos/modules/system/boot/loader/grub/install-grub.pl
@@ -112,6 +112,7 @@ struct(Fs => {
     device => '$',
     type => '$',
     mount => '$',
+    root => '$',
 });
 sub PathInMount {
     my ($path, $mount) = @_;
@@ -135,6 +136,7 @@ sub GetFs {
     foreach my $fs (read_file("/proc/self/mountinfo")) {
         chomp $fs;
         my @fields = split / /, $fs;
+        my $rootDirectory = $fields[3];
         my $mountPoint = $fields[4];
         my @mountOptions = split /,/, $fields[5];
 
@@ -159,7 +161,7 @@ sub GetFs {
             # so we only make this call last, when it's likely that this is the mount point we need.
             next unless -d $mountPoint;
 
-            $bestFs = Fs->new(device => $device, type => $fsType, mount => $mountPoint);
+            $bestFs = Fs->new(device => $device, type => $fsType, mount => $mountPoint, root => $rootDirectory);
         }
     }
     return $bestFs;
@@ -172,7 +174,7 @@ my $driveid = 1;
 sub GrubFs {
     my ($dir) = @_;
     my $fs = GetFs($dir);
-    my $path = substr($dir, length($fs->mount));
+    my $path = $fs->root . substr($dir, length($fs->mount));
     if (substr($path, 0, 1) ne "/") {
         $path = "/$path";
     }
